### ISSUE TITLE
First time experience

### ISSUE THREAD

**Issue by Ryuno-Ki** [bookwyrm-social/bookwyrm#2049]
Over on Mastodon, @mouse-reeve asked me for a report of my experience as first timer.
I have no experience with Social Book Reading software.
As such, I have to make sense of it.

In consequence, this issue might be split into actionable items. Feel free to convert it into a Discussion item, if you consider this more apt.

I'm using BookWyrm.social via the web browser as well as through https://apt.izzysoft.de/fdroid/index/apk/nl.privacydragon.bookwyrm (CC @PrivacyDragon)

## What I'm looking for

A replacement for [Book Catalogue](https://f-droid.org/en/2010/10/06/book-catalogue.html). I have 1k+ books here, that I would like to keep track of. Less about what I've read (and notes I want to take), but more about what I own (so I can look up when I'm in a book store to see, what I am missing in my collection) and ideally where I placed it (bookshelf, room).

If I can [scan ISBN](https://tech.lgbt/@bookwyrm/107956253604107939) that would ease my job. (That doesn't work on the Android app because of missing permissions). Alternatively, I want to look for title or author.

A secondary need is the ability to keep a wishlist. On social media I might come across books that sound interesting. I would like to remember them for the next time I find myself in a bookstore.

## Things I wish I could do

As of lately, [I'm using Mastodon](https://layer8.space/@RyunoKi) - I wish I could somehow link that account up with BookWyrm.social so that posts I publish on BookWyrm become toots in Mastodon.

Book Catalogue offered an export option (which is in the works as far as I have heard). I would like to post-process a CSV on my laptop. (As EU resident, a data takeout would be required anyway).

As I added some books, I was asked to enter dates. I cannot remember when I started reading books years ago, but couldn't figure out how to enter a book without a date.

## Confusing aspects

After creating an account, I [created shelves](https://bookwyrm.social/user/Ryuno-Ki) while I was trying to figure out how to add books to my collections. I went via the search eventually (but miss many mangas).

Opening the hamburger menu still showed it as hamburger icon. I was expecting an X or something to indicate a toogle.

## Things I noticed

Slow loading time (especially on the app, I don't have a visual feedback for loading). This is noticeable on search or navigation.

I'll add more / edit if something springs to my mind.

**Comment by PrivacyDragon** [comment]
> If I can scan ISBN that would ease my job. (That doesn't work on the Android app because of missing permissions).
When I created the Android app that feature did not yet exist. I will see if I can add camera permission as soon as I have time, though I might not be able to test it, for the instance I use, wyrms.de, is only at Bookwyrm version 0.3.0 instead of 0.3.4.

> (especially on the app, I don't have a visual feedback for loading).
Yes, I noticed it too. I will try to see if I can find a simple way to deal with that, to have an indication that the page is loading something. 

If the app has other problems or things that could be better, feel free to submit an issue on the Github repository. (https://github.com/PrivacyDragon/Bookwyrm_Android)

Note: as the app description already tells, the app is NOT an official client. 🙂

**Comment by mouse-reeve** [comment]
Here's what I'm seeing as actionable fixes, and again, let me know if you have different or additional ideas on this!

### 1. Getting your data into bookwyrm is challenging.

It sounds like importing from a CSV would be of huge benefit to you. This feature does exist, but it's hard to find (#1436 - merged but not deployed), and it doesn't current have good support for arbitrary CSV formats (#2050). In the mean time, you could try doing an import with your data from `Book Catalogue` -- if you have the patience and interest to format the CSV to have the same header names as a GoodReads export, I think you could get good results from this.

If there is a data source with an API that has good coverage of Manga, I'd be excited to add a connector for that (this is also a particular issue for books not in English). I don't know if that exists or what it might be, but I'd like to keep expanding the places where an instance can get data.

The reading dates are actually optional, but it isn't at all clear from the interface that this is the case, and it can be a hassle to remove a date if one is auto-populated (#1252). It might also be helpful to be able to use partial dates (#743). 

Search really is quite slow (#2051).

### 2. Your focus is on cataloguing books, whereas bookwyrm's focus is on tracking reading status on books

This is a bit more complex, because I think to some extent the issue stems from a difference between what I've viewed as the primary use case (socializing about what you're currently reading) and your primary use case (cataloguing and getting recommendations). That doesn't at all mean that your use case isn't a legit one for bookwyrm, that it isn't important, or that bookwyrm can't do a better job supporting it, but I think it does mean we may run into cases where the limitation is that we can't be everything to everyone. 

One thing that sounds like it could be helpful is adding some clarifying text about what shelves are. A custom "wish list" shelf might be a great fit for you -- it would show up in the dropdown next to the "want to read" button, which would make it straightforward to see a book your friend loves and add it to your shelf. A walkthrough might also be helpful here (#2052). Having the link to your shelves in the top bar might make this more clear as well.

I've used librarything for cataloguing, and one of the features that was very handy there was bulk edit controls on shelves (#1763 but also other options, like re-shelving)


### 3. Some features that would be helpful for you aren't present:
Exports is in progress! I know that's a really important one, and I'm making progress on it 🙂  #1556 

I haven't done anything related to cross-posting -- I do see people cross posting to twitter as well either manually or using a third party service, so you're clearly not the only one who'd like to see this. An option that's available right now is to boost your bookwyrm post from your mastodon account, since the posts do federate.

**Comment by mouse-reeve** [comment]
You're right that this could be a discussion! I temporarily forgot those existed, so I'll transfer it to that format
